File: app/src/main/java/com/example/testgithub/dependencyinjection/AppComponent.java
package com.example.testgithub.dependencyinjection;

import android.app.Application;

import com.example.testgithub.TestGithub;

import javax.inject.Singleton;

import dagger.BindsInstance;
import dagger.Component;
import dagger.android.AndroidInjectionModule;

@Singleton @Component(modules = {
    AndroidInjectionModule.class, MainActivityModule.class, AppModule.class
}) public interface AppComponent {

  @Component.Builder interface Builder {

    @BindsInstance
    Builder application(Application application);

    AppComponent build();
  }

  void inject(TestGithub testGithub);
}

File: app/src/main/java/com/example/testgithub/dependencyinjection/AppComponent.kt
package com.example.testgithub.dependencyinjection

import android.app.Application
import com.example.testgithub.TestGithub
import dagger.BindsInstance
import dagger.Component
import dagger.android.AndroidInjectionModule
import javax.inject.Singleton

@Singleton
@Component(
        modules =
        [AndroidInjectionModule::class,
            MainActivityModule::class,
            AppModule::class]
)
interface AppComponent {

    @Component.Builder
    interface Builder {

        @BindsInstance
        fun application(application: Application): Builder

        fun build(): AppComponent
    }

    fun inject(testGithub: TestGithub)
}
